express-server/src/routes/profile.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * User Profile Management Routes
+ *
+ * Handles progressive profiling for monday.com CRM integration.
+ * Users can update their profile information which syncs to monday.com.
+ */
+
+import { Response } from "express";
+import { RequestWithLogger } from "../types/request";
+import { z } from "zod";
+import * as firebase from "../Firebase";
+import { sendError } from "./sendError";
+import { createMondayItem } from "../services/monday";
+import { logger } from "tttc-common/logger";
+
+const profileLogger = logger.child({ module: "profile" });
+
+/**
+ * Profile update request schema
+ * All fields are optional for progressive profiling
+ */
+const profileUpdateSchema = z.object({
+  company: z.string().min(1).max(200).optional(),
```

Comment:
for these string values a single white space will pass. 
```suggestion
  company: z.string().trim().min(1).max(200).optional(),
```